dali/operators/reader/loader/webdataset_loader.h

```diff
@@ -31,7 +31,8 @@ namespace dali {
 namespace detail {
 namespace wds {
 
-constexpr char kCurrentIndexVersion[] = "v1.1";
+const std::string kCurrentIndexVersion = "v1.2";  // NOLINT
+const std::vector<std::string> kLegacyIndexVersions = {"v1.1"};
```

Comment:
Why not have _all_ supported versions here and make it a `set` or `unordered_set`?
`std::unordered_set<std::string> kSupportedIndexVersions = { "v1.1", "v1.2" };`
With that you could just do `kSupportedIndexVersions.count(version)` instead of running `accumulate` which is much more convoluted a way to express the same idea.

Reply:
Done